zebrad/tests/acceptance.rs

```diff
@@ -1473,3 +1487,387 @@ where
+        "Chain not synchronized to a block after the specified height"
+    );
+
+    let mut target_height = height.0;
+    let mut transactions = Vec::new();
+
+    while transactions.is_empty() {
+        transactions =
+            load_transactions_from_block(block::Height(target_height), &mut state).await?;
+
+        target_height += 1;
+    }
+
+    Ok(transactions)
+}
+
+/// Performs a request to the provided read-only `state` service to fetch all transactions from a
+/// block at the specified `height`.
+async fn load_transactions_from_block<ReadStateService>(
+    height: block::Height,
+    state: &mut ReadStateService,
+) -> Result<Vec<Arc<Transaction>>>
+where
+    ReadStateService: Service<
+        zebra_state::ReadRequest,
+        Response = zebra_state::ReadResponse,
+        Error = zebra_state::BoxError,
+    >,
+{
+    let request = zebra_state::ReadRequest::Block(HashOrHeight::Height(height));
+
+    let response = state
+        .ready()
+        .and_then(|ready_service| ready_service.call(request))
+        .map_err(|error| eyre!(error))
+        .await?;
+
+    let block = match response {
+        zebra_state::ReadResponse::Block(Some(block)) => block,
+        zebra_state::ReadResponse::Block(None) => {
+            panic!("Missing block at {height:?} from state")
+        }
+        _ => unreachable!("Incorrect response from state service: {response:?}"),
+    };
+
+    Ok(block.transactions.to_vec())
+}
+
+/// Starts a state service using the provided `cache_dir` as the directory with the chain state.
+async fn start_state_service(
+    network: Network,
+    cache_dir: impl Into<PathBuf>,
+) -> Result<(
+    BoxStateService,
+    impl Service<
+        zebra_state::ReadRequest,
+        Response = zebra_state::ReadResponse,
+        Error = zebra_state::BoxError,
+    >,
+    LatestChainTip,
+    ChainTipChange,
+)> {
+    let config = zebra_state::Config {
+        cache_dir: cache_dir.into(),
+        ..zebra_state::Config::default()
+    };
+
+    remove_file_if_it_exists(config.db_path(network).join("LOCK")).await?;
+
+    Ok(zebra_state::init(config, network))
+}
+
+/// Spawns a zebrad instance to interact with lightwalletd, but without an internet connection.
+///
+/// This prevents it from downloading blocks. Instead, the `zebra_directory` parameter allows
+/// providing an initial state to the zebrad instance.
+fn spawn_zebrad_for_rpc_without_initial_peers(
+    zebra_directory: TempDir,
+) -> Result<(TestChild<TempDir>, SocketAddr)> {
+    let mut config = random_known_rpc_port_config()
+        .expect("Failed to create a config file with a known RPC listener port");
+
+    config.state.ephemeral = false;
+    config.network.initial_mainnet_peers = HashSet::new();
+    config.network.initial_testnet_peers = HashSet::new();
+
+    let mut zebrad = zebra_directory
+        .with_config(&mut config)?
+        .spawn_child(&["start"])?
+        .with_timeout(Duration::from_secs(60 * 60));
```

Comment:
Question: What timeout should we have for `zebrad`?

Reply:
I added a `FINISH_PARTIAL_SYNC_TIMEOUT` constant for one of the Zebrad instances and a `LIGHTWALLETD_TEST_TIMEOUT` constant for the other Zebrad instance and for the lightwalletd instance.

We can tweak the values if we want, I just added values that worked for me.